rust/index/src/metadata/types.rs

```diff
@@ -506,7 +509,10 @@ impl<'me> MetadataIndexReader<'me> {
                         }
                         let rbm = blockfile_reader.get(metadata_key, k).await;
                         match rbm {
-                            Ok(rbm) => Ok(rbm),
+                            Ok(Some(rbm)) => Ok(rbm),
+                            Ok(None) => Err(MetadataIndexError::BlockfileError(Box::new(
+                                BlockfileError::NotFoundError,
+                            ))),
```

Comment:
I'm wondering if we should return `Ok(<empty bitmap>)` instead of an error here? I suppose if the user uses non-existent `(key, value)` pair in a where clause, we may not want to yield an error for the query.

Reply:
Same for all following occurances

Reply:
At a closer look the previous `if` clause should make sure the `Ok(None)` branch is never reachable. Should we remove the `if` clause or mark this branch with `unreachable!`?

Reply:
I'd prefer to leave it to return an error if the branch is reached, and leave the if so we never error.  How would you feel about that?